SwaggerGen/Swagger/Type/Custom/EmailType.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace SwaggerGen\Swagger\Type\Custom;
+
+/**
+ * Email type definition
+ *
+ * @package    SwaggerGen
+ * @author     Martijn van der Lee <martijn@vanderlee.com>
+ * @copyright  2014-2017 Martijn van der Lee
+ * @license    https://opensource.org/licenses/MIT MIT
+ */
+class EmailType extends \SwaggerGen\Swagger\Type\StringType implements ICustomType
+{
+
+	const PATTERN = '\A[a-z0-9!#$%&\'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&\'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\z';
+	const TYPE = 'email';
+
+	public static function register($type = null)
+	{
+		self::registerType($type === null ? self::TYPE : $type, __CLASS__);
```

Comment:
I don't think registering same class under many names is actually all that useful. Do you have any specific case in mind?